File: fm-domain/src/main/java/io/zeitwert/fm/contact/adapter/api/jsonapi/impl/ObjContactApiRepositoryImpl.java
package io.zeitwert.fm.contact.adapter.api.jsonapi.impl;

import io.zeitwert.dddrive.app.model.RequestContext;
import io.zeitwert.dddrive.ddd.adapter.api.jsonapi.base.AggregateApiRepositoryBase;
import io.zeitwert.fm.contact.adapter.api.jsonapi.ObjContactApiRepository;
import io.zeitwert.fm.contact.adapter.api.jsonapi.dto.ObjContactDto;
import io.zeitwert.fm.contact.model.ObjContact;
import io.zeitwert.fm.contact.model.ObjContactRepository;
import io.zeitwert.fm.oe.model.ObjUserRepository;
import org.springframework.stereotype.Controller;

@Controller("objContactApiRepository")
public class ObjContactApiRepositoryImpl
		extends AggregateApiRepositoryBase<ObjContact, ObjContactDto>
		implements ObjContactApiRepository {

	public ObjContactApiRepositoryImpl(
			ObjContactRepository repository,
			RequestContext requestCtx,
			ObjUserRepository userRepository,
			ObjContactDtoAdapter dtoAdapter) {
		super(ObjContactDto.class, requestCtx, userRepository, repository, dtoAdapter);
	}

}

File: fm-domain/src/main/java/io/zeitwert/fm/contact/adapter/api/jsonapi/impl/ObjContactApiRepositoryImpl.kt
package io.zeitwert.fm.contact.adapter.api.jsonapi.impl

import dddrive.ddd.core.model.RepositoryDirectory
import io.zeitwert.dddrive.app.model.SessionContext
import io.zeitwert.dddrive.ddd.adapter.api.jsonapi.base.GenericAggregateApiRepositoryBase
import io.zeitwert.fm.contact.adapter.api.jsonapi.dto.ObjContactDto
import io.zeitwert.fm.contact.model.ObjContact
import io.zeitwert.fm.contact.model.ObjContactRepository
import org.springframework.stereotype.Controller

/**
 * Generic API repository for ObjContact using the GenericAggregateDtoAdapter.
 *
 * This replaces ObjContactApiRepositoryImpl and uses metadata-driven serialization
 * instead of manual DTO mapping.
 */
@Controller("objContactApiRepository")
open class ObjContactApiRepositoryImpl(
	directory: RepositoryDirectory,
	repository: ObjContactRepository,
	adapter: ObjContactDtoAdapter,
	sessionCtx: SessionContext,
) : GenericAggregateApiRepositoryBase<ObjContact, ObjContactDto>(
		resourceClass = ObjContactDto::class.java,
		directory = directory,
		repository = repository,
		adapter = adapter,
		sessionCtx = sessionCtx,
	)
